File: sdk/node/test_live.js
const HelmAnalytics = require('helm-analytics');

// Use the same Site ID we used for Python
const SITE_ID = "e3d3d060-dc0d-4301-894a-5994f65e2216";

console.log(`Initializing Helm Analytics (Node.js) with Site ID: ${SITE_ID}`);
const helm = new HelmAnalytics({ siteId: SITE_ID });

const mockReq = {
    originalUrl: 'http://test-node-sdk.com/integration-test',
    url: 'http://test-node-sdk.com/integration-test',
    headers: {
        'user-agent': 'Helm-Node-SDK-Test/1.0',
        'referer': 'http://npmjs.org',
        'x-forwarded-for': '5.6.7.8'
    },
    socket: { remoteAddress: '5.6.7.8' }
};

console.log("Sending test event: 'node_sdk_test'...");

// track() is async but fires-and-forgets internally usually, 
// strictly speaking it returns a promise we can await if we want validation.
helm.track(mockReq, 'pageview', { status: 'success' })
    .then(() => {
        console.log("Event dispatched! Check your dashboard.");
        // Short delay to ensure network flush
        setTimeout(() => process.exit(0), 2000);
    })
    .catch(err => {
        console.error("Tracking Error:", err);
    });

File: sdk/python/test_live.py
from helm_analytics import HelmAnalytics
import time
import sys

# Use the Site ID you were using for funnels
SITE_ID = "e3d3d060-dc0d-4301-894a-5994f65e2216"

def test_sdk():
    print(f"Initializing Helm Analytics with Site ID: {SITE_ID}")
    helm = HelmAnalytics(site_id=SITE_ID)
    
    # Mocking a request object for testing
    class MockRequest:
        url = "http://test-python-sdk.com/integration-test"
        remote_addr = "1.2.3.4"
        headers = {
            "User-Agent": "Helm-Python-SDK-Test/1.0",
            "Referer": "http://pypi.org"
        }

    print("Sending test event: 'pageview'...")
    helm.track(MockRequest(), event_type="pageview", metadata={"status": "success"})
    
    # Wait a bit for the background thread to fire
    time.sleep(2)
    print("Event sent! Check your dashboard.")

if __name__ == "__main__":
    try:
        test_sdk()
    except ImportError:
        print("Error: helm_analytics package not found.")
        print("Please run: pip install helm-analytics")
